stock_by_warehouse_sale/models/sale_order_line.py

```diff
@@ -0,0 +1,20 @@
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    warehouses_stock = fields.Json(compute="_compute_warehouse_stock")
+    warehouses_stock_recompute = fields.Boolean(store=False)
+
+    def _compute_get_warehouses_stock(self):
+        for line in self:
+            line.warehouses_stock = line.product_id.with_context(
```

Comment:
Is it easier

```suggestion
            line.warehouses_stock = line.product_id.with_context(warehouse_id=line.warehouse_id).warehouses_stock
```

Reply:
It doesn't work since the field in the product is not yet calculated and is not stored so is need to call the compute directly.

Reply:
True, sorry